sample/trace_key.go

```diff
@@ -42,16 +43,29 @@ func (d *traceKey) build(trace *types.Trace) string {
 	// if they happen multiple times.
 	fieldCollector := make(map[string][]string)
 
+	// once a key gets this many unique values, it's off the charts in terms of uniqueness
+	// so we just stop looking for more.
+	// This is a safety valve to prevent us from someone sending a high-cardinality field
+	// in a giant trace.
+	const maxKeyLength = 100
 	// for each field, for each span, get the value of that field
 	spans := trace.GetSpans()
+	uniques := generics.NewSetWithCapacity[string](maxKeyLength)
+outer:
 	for _, field := range d.fields {
 		for _, span := range spans {
 			if val, ok := span.Data[field]; ok {
-				fieldCollector[field] = append(fieldCollector[field], fmt.Sprintf("%v", val))
+				v := fmt.Sprintf("%v", val)
+				uniques.Add(v)
+				if len(uniques) >= maxKeyLength {
```

Comment:
We don't use the uniques set for anything other than counting, could we just use simple integer counter instead?

Reply:
I had done it that way originally, but the problem is that it's OK to have 1000 instances of a low-cardinality field since we only keep one of them. I just noticed, though, that fieldCollector can still get really large in that situation, so I've just added some code that actually checks that we have a new value before we add it, which will prevent fieldCollector from getting big.